### PR TITLE
Code changes for JSON Record stream without any custom formats

### PR DESCRIPTION
**Pull request by sadasivaprasad-trellix**

### CODE REVIEW

**Review by umanekar-mcafee — CHANGES_REQUESTED** [review]

File: src/main/java/com/opendxl/databus/common/internal/adapter/DatabusProducerJSONRecordAdapter.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        @Override
+        public org.apache.kafka.clients.producer.ProducerRecord<String, byte[]> adapt(
+                        final ProducerRecord sourceProducerRecord) {
+
+                Map<String, String> headers = sourceProducerRecord.getHeaders().getAll();
+                final List<Header> kafkaHeaders = new ArrayList<>();
+
+                for (final String key : headers.keySet()) {
+                        String val = headers.get(key);
+                        if (null == val) {
```
Comment (gmishra-mcafee) [inline src/main/java/com/opendxl/databus/common/internal/adapter/DatabusProducerJSONRecordAdapter.java:64]: If we are not getting value for a key then why we are adding empty value for that key ?

**Review by gmishra-mcafee — CHANGES_REQUESTED** [review]

File: src/main/java/com/opendxl/databus/common/internal/adapter/DatabusProducerJSONRecordAdapter.java (hunk 1 above)
Comment (sadasivaprasad-trellix) [reply src/main/java/com/opendxl/databus/common/internal/adapter/DatabusProducerJSONRecordAdapter.java]: The null check is removed as the Headers class  checks for null and inserts empty value

File: src/main/java/com/opendxl/databus/consumer/Consumer.java [hunk 2]
```diff
@@ -78,6 +79,11 @@ public abstract class Consumer<P> {
      */
     protected Map<String, Object> headerFilter;
 
+    /**
```
[the remark this answers is not in the document]
Comment (sadasivaprasad-trellix) [reply src/main/java/com/opendxl/databus/consumer/Consumer.java]: Removed.

File: src/main/java/com/opendxl/databus/common/internal/adapter/ConsumerJSONRecordAdapter.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import org.apache.kafka.common.header.Header;
+
+/**
+ * Adapter for ConsumerRecord.
+ */
+public final class ConsumerJSONRecordAdapter<P> implements
+        Adapter<org.apache.kafka.clients.consumer.ConsumerRecord<String, byte[]>,
+                ConsumerRecord<P>> {
+
+    class TargetRecord {
```
[the remark this answers is not in the document]
Comment (sadasivaprasad-trellix) [reply src/main/java/com/opendxl/databus/common/internal/adapter/ConsumerJSONRecordAdapter.java:19]: This was intended to keep all the required fields within the class scope. Also Ulhas has commented to use an inner class to keep it neat.

**Review by gmishra-mcafee — APPROVED** [review]

**Pull request merged**